components/net/indexeddb/idb_thread.rs

```diff
@@ -0,0 +1,343 @@
+                        let store_name = SanitizedName::new(store_name);
+                        self.get_database_mut(origin, db_name).map(|db| {
+                            // Queues an operation for a transaction without starting it
+                            db.queue_operation(sender, store_name, txn, mode, operation);
+                            // FIXME:(arihant2math) Schedule transactions properly:
+                            // for now, we start them directly.
+                            db.start_transaction(txn, None);
+                        });
+                    },
+                }
+            }
+        }
+    }
+
+    fn get_database(
+        &self,
+        origin: ImmutableOrigin,
+        db_name: String,
+    ) -> Option<&IndexedDBEnvironment<HeedEngine>> {
+        let idb_description = IndexedDBDescription {
+            origin,
+            name: db_name,
+        };
+
+        self.databases.get(&idb_description)
+    }
+
+    fn get_database_mut(
+        &mut self,
+        origin: ImmutableOrigin,
+        db_name: String,
+    ) -> Option<&mut IndexedDBEnvironment<HeedEngine>> {
+        let idb_description = IndexedDBDescription {
+            origin,
+            name: db_name,
+        };
+
+        self.databases.get_mut(&idb_description)
+    }
+
+    fn handle_sync_operation(&mut self, operation: SyncOperation) {
+        match operation {
+            SyncOperation::OpenDatabase(sender, origin, db_name, version) => {
+                let idb_description = IndexedDBDescription {
+                    origin,
+                    name: db_name,
+                };
+
+                let idb_base_dir = self.idb_base_dir.as_path();
+
+                match self.databases.entry(idb_description.clone()) {
+                    Entry::Vacant(e) => {
+                        let db = IndexedDBEnvironment::new(
+                            HeedEngine::new(idb_base_dir, &idb_description.as_path()),
+                            version.unwrap_or(0),
+                        );
+                        sender.send(db.version).unwrap();
+                        e.insert(db);
+                    },
+                    Entry::Occupied(db) => {
+                        sender.send(db.get().version).unwrap();
+                    },
+                }
+            },
+            SyncOperation::DeleteDatabase(sender, origin, db_name) => {
+                let idb_description = IndexedDBDescription {
+                    origin,
+                    name: db_name,
+                };
+                self.databases.remove(&idb_description);
+
+                // FIXME:(rasviitanen) Possible security issue?
+                let mut db_dir = self.idb_base_dir.clone();
+                db_dir.push(&idb_description.as_path());
+                if std::fs::remove_dir_all(&db_dir).is_err() {
+                    sender.send(Err(())).unwrap();
+                } else {
+                    sender.send(Ok(())).unwrap();
```

Comment:
```suggestion
                    let _ = sender.send(Err(()));
                } else {
                    let _ = sender.send(Ok(()));
```

Reply:
Let's make similar changes to the other uses of unwrap/expect on send calls in this file.